src/towncrier/test/test_build.py

```diff
@@ -568,55 +607,79 @@ def test_singlefile_errors_and_explains_cleanly(self):
             result.output,
         )
 
-    def test_single_file_false(self):
+    def test_single_file(self):
         """
-        If formatting arguments are given in the filename arg and single_file is
-        false, the filename will not be formatted.
+        "If a single file is used, the content of that file gets overwritten each time."
         """
         runner = CliRunner()
 
-        with runner.isolated_filesystem():
-            with open("pyproject.toml", "w") as f:
-                f.write(
-                    '[tool.towncrier]\n single_file=false\n filename="{version}-notes.rst"'
-                )
-            os.mkdir("newsfragments")
-            with open("newsfragments/123.feature", "w") as f:
-                f.write("Adds levitation")
+        def do_build_once_with(version, fragment_file, fragment):
+            with open(f"newsfragments/{fragment_file}", "w") as f:
+                f.write(fragment)
 
             result = runner.invoke(
                 _main,
                 [
                     "--version",
-                    "7.8.9",
+                    version,
                     "--name",
                     "foo",
                     "--date",
                     "01-01-2001",
                     "--yes",
                 ],
             )
+            # not git repository, manually remove fragment file
+            Path(f"newsfragments/{fragment_file}").unlink()
+            return result
 
-            self.assertEqual(0, result.exit_code, result.output)
+        results = []
+        with runner.isolated_filesystem():
+            with open("pyproject.toml", "w") as f:
+                f.write(
+                    '[tool.towncrier]\n single_file=true\n filename="{version}-notes.rst"'
+                )
```

Comment:
I think that this is the important part.
```suggestion
                f.write(
                    '[tool.towncrier]\n'
                    ' single_file=true\n'
                    ' # The `filename` variable is fixed and not formated\n`
                    ' # in any way.\n'
                    ' filename="{version}-notes.rst"\n
                    ''
                )
```